src/index.ts

```diff
@@ -122,31 +135,34 @@ const scrubCoin = (x: Coin) => {
 
 const scrubCoins = (x: readonly Coin[]) => x.filter(c => c.amount).map(scrubCoin)
 
-type Route = { poolId: unknown; tokenOutDenom: unknown }
+type Route = { pool_id: unknown; token_out_denom: unknown }
 
 const scrubRoute = (x: Route) => {
-  if (!x.poolId) throw new Error('missing route poolId')
-  if (!x.tokenOutDenom) throw new Error('missing route tokenOutDenom')
+  if (!x.pool_id) throw new Error('missing route pool_id')
+  if (!x.token_out_denom) throw new Error('missing route token_out_denom')
 
   return {
-    poolId: x.poolId,
-    tokenOutDenom: x.tokenOutDenom
+    poolId: x.pool_id,
+    tokenOutDenom: x.token_out_denom
   }
 }
 
 const scrubRoutes = (x: Route[]) => x.map(scrubRoute)
 
-function parse_legacy_tx_format({ fee, memo, msg, signatures }: amino.StdTx): ProtoTx {
-  if (msg.length !== 1) throw new Error('multiple msgs not supported!')
+function parse_legacy_tx_format(tx: AgnosticStdTx): ProtoTx {
+  const msgOrMsgs = tx.msg ?? tx.msgs
```

Comment:
union type would allow proper narrowing here

Reply:
`const getMessagesAmino = (tx: amino.StdTx | StdTxWithMsgs): readonly amino.AminoMsg[] => 'msgs' in tx ? tx.msgs : tx.msg`